### DISCUSSION TITLE
5.5.0 Release Planning

### DISCUSSION THREAD

**Discussion by dangotbanned** [vega/altair#3687] [General]
Thought I'd kick this off as we've had some questions on when a `5.5.0` release is expected:

- https://github.com/vega/altair/issues/3684#issuecomment-2482912997
- https://github.com/vega/altair/issues/3620#issuecomment-2489833450

The primary user-facing concern of `3.13` support has been resolved, but [only](https://github.com/vega/altair/issues/3684#issuecomment-2482833786) on `main`
- #3591

I had previously collected some issues/PRs in (https://github.com/vega/altair/pull/3591#issuecomment-2437995034) and the remaining ones are all part of:
- #3519

I'll follow this up with some threads on other topics/unresolved conversations.

For the last month we have [pulse](https://github.com/vega/altair/pulse/monthly) but there may be some older items of interest.

cc @mattijn @jonmmease @joelostblom @binste @ChristopherDavisUCI

**Comment by jonmmease** [comment]
I'd like to get a release out this weekend at the latest. I haven't done one yet, but I should have the permission.

Is anything in an incomplete state now where we wouldn't want to release as-is? For feature work we can follow on with another release as soon as we are ready.

**Reply by mattijn** [reply]
I've added #3689, #3690 and #3691.
I was wondering about [this line](https://github.com/vega/altair/blob/main/pyproject.toml#L66) in the pyproject.toml part of `all`:
```cmd
"vegafusion[embed]>=1.6.6",
```
Is this still valid for vegafusion v2.0? Or should there be an upper bound?

Good luck with the release! If you need help, just ping and I will try my utmost.

**Reply by jonmmease** [reply]
Thanks @mattijn, yes that vegafusion lower bound is still valid, but altiar 5.5.0 will be forward compatible with vegafusion 2.

I'll plan to do this on Saturday. I'm not sure yet when in the day, so let me know if anyone is working on anything into tomorrow.

**Reply by dangotbanned** [reply]
@jonmmease my only concerns would be regarding docs.

@mattijn, @joelostblom how do you feel about this **not** making it in before `5.5.0`?
I would consider this a bit of a special case since it would explain the need for the deprecation (#3618).

If anyone has time **today** to give some input on *what* might need explaining - I could still probably get it written up this weekend.
I really need a perspective from someone who has *read less* of https://typing.readthedocs.io/en/latest/spec/index.html

- #3645
	- @mattijn (https://github.com/vega/altair/issues/3645#issuecomment-2493829661) 

---

Another one is an upstream issue in the docs, but may be fine to release as-is.

@jonmmease could you try building the docs and see how you feel about the responsiveness on [chart-themes](https://altair-viz.github.io/user_guide/customization.html#chart-themes)?
There is a pretty buggy feel to the page before `pyodide` has fully loaded

- #3642
	- https://github.com/vega/altair/pull/3630#discussion_r1801090949
	- #3630

---

If those two aren't an issue for anyone else, I'm happy to release asap

**Reply by jonmmease** [reply]
Thanks Dan,

I honestly didn't follow the theme work closely.  Are the current contents of user_guide/customization.html on `main` accurate and usable without a warning?  If so then I think that's enough to move forward with. I don't think we need to block on additional documentation if the basic level of docs that we already had are still valid.  There is also the option of removing the deprecation warning and advertising the new API as experimental if you were more comfortable with that.

I built the theme docs locally and I do see what you mean about there being a bit of lag. I don't think it's blocking, but it might be nice to move this to a dedicated page so that you don't need to load pyodide to look at the theme API. But the theme explorer is really nice!

Something else that comes to mind about the docs, I don't think we necessarily need to fully couple docs updates with releases the way we have been. I don't see why we couldn't keep iterating on the docs after the 5.5.0 release (prior to a 5.5.1 or 5.6.0 release).

I started a HackMD document for the release notes at https://hackmd.io/@jEkhP0lORMCL8z1MH2P2SA/S188R-0zkg. I invited @dangotbanned, @mattijn, and @joelostblom as editors (3 seems to be the limit for for the free plan). Right now, it's just a copy of the GitHub auto-generated release notes, but I'll review and work on organizing tomorrow. @dangotbanned would you want to add a small section on themes here that highlights the deprecation and includes links to the docs?

Thanks all!

**Reply by dangotbanned** [reply]
> Are the current contents of user_guide/customization.html on `main` accurate and usable without a warning?

@jonmmease yes to both of those, so this isn't blocking for me - especially given that `5.5.0` will help out downstream dependencies.

Just wanted to make sure everyone had a chance to consider before a release.

@mattijn has made a great start on the docs in (https://github.com/vega/altair/issues/3645#issuecomment-2493829661) so I'm confident we can iterate on that *after* the release

**Reply by joelostblom** [reply]
> @mattijn, @joelostblom how do you feel about this not making it in before 5.5.0?

Ok from my side of things, since we can follow up with a doc update soon after.

> I don't think we need to block on additional documentation if the basic level of docs that we already had are still valid.

+1

> Something else that comes to mind about the docs, I don't think we necessarily need to fully couple docs updates with releases the way we have been. I don't see why we couldn't keep iterating on the docs after the 5.5.0 release (prior to a 5.5.1 or 5.6.0 release).

Partly agree here. I think it depends on the type of change in the docs. If it is just clarifications of existing functionality then I agree, but we should be careful with functional changes, so that the docs are not out of sync with the functionality available in the latest installable version of altair.

**Reply by jonmmease** [reply]
Yes, that's a good point regarding between release doc updates. We would need to be careful that they apply only to the current released version.

**Reply by dangotbanned** [reply]
Maybe we could use [Milestones](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) for tracking things like this?

I added some new labels somewhat recently, but they wouldn't be as useful for purely new features

- https://github.com/vega/altair/pull/3623

**Comment by dangotbanned** [comment]
Finding out now that a new `ruff` has landed:
- https://github.com/astral-sh/ruff/releases/tag/0.8.0

Haven't read through the notes yet, but assuming `RUF200` came out of preview

https://github.com/vega/altair/actions/runs/11976814528/job/33393326277?pr=3631

This could be something we wanna fix before a release:

```
pyproject.toml:37:17: RUF200 Failed to parse pyproject.toml: invalid type: map, expected a sequence
   |
35 | requires-python = ">=3.9"
36 | dynamic = ["version"]
37 | license-files = { paths = ["LICENSE"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^ RUF200
38 | classifiers = [
39 |     "Development Status :: 5 - Production/Stable",
   |
```

I'd forgotten about this, but I disabled my [*local* validation](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml) of `pyproject.toml` a while back because of this issue.

Is there any reason we'd want to keep `license-files` in this format?

**Reply by dangotbanned** [reply]
There are some examples of how we could correct this in https://peps.python.org/pep-0639/#add-license-files-key

**Reply by dangotbanned** [reply]
Resolved in
- https://github.com/vega/altair/pull/3693#discussion_r1854405044

**Comment by jonmmease** [comment]
I finished the release notes at https://hackmd.io/@jEkhP0lORMCL8z1MH2P2SA/S188R-0zkg and I'm starting on the release. A couple of notes.

Steps 8 and 12 of [RELEASING.md](https://github.com/vega/altair/blob/main/RELEASING.md?plain=1) include instructions for tagging and pushing the release tag. I'm planning to remove step 8, so that the tag comes after publishing the packages.

When I build the docs after cleaning, I'm getting some sphinx errors and warnings:

```
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/api.rst:260:<autosummary>:1: ERROR: Unknown target name: "generic". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cumulative distribution function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cumulative distribution function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cumulative distribution function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "probability density function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "probability density function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "probability density function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "continuous color scheme". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "hcl". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "hsl". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cie lab". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "d3-color's rgb function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cumulative distribution function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cumulative distribution function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "cumulative distribution function". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "rgb". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "log-normal probability distribution". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "normal (gaussian) probability distribution". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "continuous uniform probability distribution". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "window.screen". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "*unit*". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "*unit*". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "*units*". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "utc". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "*unit*". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "utc". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/generated/api-cls/altair.expr.rst:182:<autosummary>:1: ERROR: Unknown target name: "*unit*". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.Axis:284: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.Axis:284: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.AxisConfig:287: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.AxisConfig:287: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.Encoding:92: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.FacetedEncoding:100: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.LegendConfig:169: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.LegendConfig:169: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.SharedEncoding:59: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.AxisConfigKwds:287: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.AxisConfigKwds:287: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.LegendConfigKwds:169: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.LegendConfigKwds:169: WARNING: Inline literal start-string without end-string. [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_typing.py:docstring of altair.vegalite.v5.schema._typing.RowColKwds:2: ERROR: Unknown target name: "generic". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.EncodeKwds:100: WARNING: Inline literal start-string without end-string. [docutils]
```

I'm just starting to look into these now, but let me know if others have seen this.

**Reply by jonmmease** [reply]
looks like the errors are from sphinx not understanding target references in docstrings.

So 

```
        """
        Trigonometric sine.

        Same as JavaScript's `Math.sin`_.

        .. _Math.sin: \
            https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sin
        """
```

is not working, but

```
        """
        Trigonometric sine.

        Same as JavaScript's `Math.sin <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sin>`__.
        """
```
is. I'm going to make these changes in the release branch, but happy to change it back later if someone figures out what the issue is with this style.

**Reply by dangotbanned** [reply]
IIRC the `docutils` errors didn't prevent `sphinx` from rendering correctly

**Reply by dangotbanned** [reply]
These one's I think were docstring line-wrapping in the middle of links.
But `sphinx` seemed to still produce the links

> ```
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.Axis:284: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.Axis:284: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.AxisConfig:287: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.AxisConfig:287: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.Encoding:92: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.FacetedEncoding:100: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.LegendConfig:169: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.LegendConfig:169: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/core.py:docstring of altair.vegalite.v5.schema.core.SharedEncoding:59: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.AxisConfigKwds:287: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.AxisConfigKwds:287: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.LegendConfigKwds:169: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_config.py:docstring of altair.vegalite.v5.schema._config.LegendConfigKwds:169: WARNING: Inline literal start-string without end-string. [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.EncodeKwds:100: WARNING: Inline literal start-string without end-string. [docutils]
> ```

**Reply by dangotbanned** [reply]
These are using `typing.Generic`.

No idea why `docutils` seems to care about that though

> ```
> /Users/jonmmease/VegaFusion/repos/altair/doc/user_guide/api.rst:260:<autosummary>:1: ERROR: Unknown target name: "generic". [docutils]
> /Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/_typing.py:docstring of altair.vegalite.v5.schema._typing.RowColKwds:2: ERROR: Unknown target name: "generic". [docutils]

**Reply by dangotbanned** [reply]
All the links are relating to (https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/altair/expr/__init__.py)

But I did quite a bit of work to get them rendered correctly here:

https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/tools/vega_expr.py#L252-L278

https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/tools/vega_expr.py#L852-L889

**Reply by jonmmease** [reply]
Thanks @dangotbanned, I really like this generation approach for the expression docstrings!

When I run `hatch run doc:build-html` and see these target errors, I'm seeing broken links in the http://localhost:8000/user_guide/generated/api-cls/altair.expr.html page. The backticks and underscore are displayed and the link doens't go anywhere.

![Screenshot 2024-11-23 at 10 34 10 AM](https://github.com/user-attachments/assets/00f275a9-95a9-40c0-98c4-051f986b8c1b)

Does this render properly for you? I don't remember, was there a particular reason you didn't want to use inline links (these are rendering properly for me)?

Would you be alright with me switching to use inline links for the release today, and then we can circle back to sort out what's going on later?

**Reply by dangotbanned** [reply]
> Thanks @dangotbanned, I really like this generation approach for the expression docstrings!

Thanks @jonmmease!

My motivation was that I found the docs very difficult to read in an editor when using inline links.
These were the docstrings before the update:
- https://github.com/vega/altair/blob/f2ac0a14294da11cd29cf57753d774f9a71ff173/altair/expr/__init__.py

> When I run `hatch run doc:build-html` and see these target errors, I'm seeing broken links in the http://localhost:8000/user_guide/generated/api-cls/altair.expr.html page. The backticks and underscore are displayed and the link doens't go anywhere.
> Does this render properly for you? I don't remember, was there a particular reason you didn't want to use inline links (these are rendering properly for me)?
> 
> Would you be alright with me switching to use inline links for the release today, and then we can circle back to sort out what's going on later?

I'll take a look now, just need to rebuild the docs locally.

I didn't know about the warnings *actually* leading to issues 😞 

### Edit

They definitely work in VSCode, although I know that isn't the issue here

![image](https://github.com/user-attachments/assets/7b9cf7c1-1af8-46d8-bceb-32d8886fa047)

This seems to be valid [reStructuredText](https://docutils.sourceforge.io/docs/user/rst/quickref.html#hyperlink-targets)

```rst
Returns the value of the `cumulative distribution function`_ at the given input domain ``value`` for a log-normal distribution with specified log ``mean`` and log standard deviation ``stdev``.

If unspecified, the log mean defaults to ``0`` and the log standard deviation defaults to
``1``.

.. _cumulative distribution function:
    https://en.wikipedia.org/wiki/Cumulative_distribution_function
```

**Reply by jonmmease** [reply]
> My motivation was that I found the docs very difficult to read in an editor when using inline links.  

yeah, that makes sense. And you're approach makes good sense! Just not sure why sphinx doesn't follow. I'll keep playing with it as well.

**Reply by dangotbanned** [reply]
> Does this render properly for you?

No I'm getting the same as you @jonmmease, the targets for the links are all like this:

- http://localhost:8000/user_guide/generated/api-cls/altair.expr.html#id1


I think the issue is that the text displayed here is the just the first line of the doc - but the link targets are never defined on the first line.

![image](https://github.com/user-attachments/assets/f83a9c70-21aa-4267-a8a5-a2227a02c137)


So the `.rst` is valid, but [autosummary](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html) doesn't appear to keep the full context of the docstring - if that makes sense?

**Reply by dangotbanned** [reply]
I find the `sphinx` docs quite a challenging read - but will see if I can find anything that might help this

**Reply by jonmmease** [reply]
> if that makes sense?

ahh, yes it does. I tried putting the link definition at the start of the docstring.

```python
    @classmethod
    def cumulativeLogNormal(
        cls,
        value: IntoExpression,
        mean: IntoExpression = None,
        stdev: IntoExpression = None,
        /,
    ) -> Expression:
        """
        .. _cumulative distribution function: https://en.wikipedia.org/wiki/Cumulative_distribution_function
        
        Returns the value of the `cumulative distribution function`_ at the given input domain ``value`` for a log-normal distribution with specified log ``mean`` and log standard deviation ``stdev``.

        If unspecified, the log mean defaults to ``0`` and the log standard deviation defaults to
        ``1``.
        """
        return FunctionExpression("cumulativeLogNormal", (value, mean, stdev))

```

The `cumulativeLogNormal` is then empty in the table, but the other functions that reference `cumulative distribution function` render properly (like `cumulativeUniform`).

![Screenshot 2024-11-23 at 11 35 10 AM](https://github.com/user-attachments/assets/95768afa-0cf4-4996-ae28-ed5256cfd989)

So I think we either need to get autosummary to include the whole docstring (which I don't see an option for in https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html), or we need to include the links in `doc/user_guide/generated/api-cls/altair.expr.rst:`.

**Reply by dangotbanned** [reply]
Maybe we could collect the targets and define them on the [altair.expr](https://altair-viz.github.io/user_guide/generated/api-cls/altair.expr.html#altair.expr) page?

Something like this

https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/doc/user_guide/large_datasets.rst?plain=1#L465-L470


### Edit
> or we need to include the links in doc/user_guide/generated/api-cls/altair.expr.rst

You beat me to it @jonmmease 😉

**Reply by jonmmease** [reply]
ahh, got it. If we include the link definitions them in the `expr` _class_ docstring, then they are available in the method docstrings.

**Reply by jonmmease** [reply]
I've got a PR almost finished

**Reply by dangotbanned** [reply]
> I've got a PR almost finished

Oh perfect, I was working on it as well but I'll leave it to you

If you need anything solved to get it over the line I can share my diff to help?

**Reply by dangotbanned** [reply]
All I had left was deduplicating the links - but probably could've avoided that by approaching it differently

**Reply by jonmmease** [reply]
I'll tag tag you in the PR soon!  After this fix, then text sphinx error I'm getting is:

```
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.Radius:127: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.RadiusDatum:41: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.Theta:127: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.ThetaDatum:41: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.X:146: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.XDatum:60: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.Y:146: ERROR: Unknown interpreted text role "raw-html". [docutils]
/Users/jonmmease/VegaFusion/repos/altair/altair/vegalite/v5/schema/channels.py:docstring of altair.vegalite.v5.schema.channels.YDatum:60: ERROR: Unknown interpreted text role "raw-html". [docutils]
```

The stack portion of the docstring for `YDatum` looks like this

```
    stack : bool, :class:`StackOffset`, Literal['zero', 'center', 'normalize'], None
        Type of stacking offset if the field should be stacked. ``stack`` is only applicable
        for ``x``, ``y``, ``theta``, and ``radius`` channels with continuous domains. For
        example, ``stack`` of ``y`` can be used to customize stacking for a vertical bar
        chart.

        ``stack`` can be one of the following values:

        * ``"zero"`` or ``true``: stacking with baseline offset at zero value of the scale
          (for creating typical stacked `bar
          <https://vega.github.io/vega-lite/docs/stack.html#bar>`__ and `area
          <https://vega.github.io/vega-lite/docs/stack.html#area>`__ chart).
        * ``"normalize"`` - stacking with normalized domain (for creating `normalized
          stacked bar and area charts
          <https://vega.github.io/vega-lite/docs/stack.html#normalized>`__ and pie charts
          `with percentage tooltip
          <https://vega.github.io/vega-lite/docs/arc.html#tooltip>`__).  :raw-html:`<br/>`
          -``"center"`` - stacking with center baseline (for `streamgraph
          <https://vega.github.io/vega-lite/docs/stack.html#streamgraph>`__).
        * ``null`` or ``false`` - No-stacking. This will produce layered `bar
          <https://vega.github.io/vega-lite/docs/stack.html#layered-bar-chart>`__ and area
          chart.
```

With the ` :raw-html:`<br/>` not being accepted. Do you remember anything about this?

**Reply by dangotbanned** [reply]
@jonmmease I'm sending from my phone, but IIRC if you search our repo for `:raw-html:` it is used in a few different places

**Reply by dangotbanned** [reply]
As in, I'm pretty sure we are interpreting it *somewhere*

**Reply by dangotbanned** [reply]
@jonmmease some examples

Also https://docutils.sourceforge.io/docs/ref/rst/roles.html#specialized-roles

https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/tools/markup.py#L37-L43

https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/tools/schemapi/codegen.py#L293-L310


https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/sphinxext/altairgallery.py#L64-L70


https://github.com/vega/altair/blob/17aeb2c837717b04563d06e2c326276ef6e7746d/doc/user_guide/interactions/jupyter_chart.rst?plain=1#L65-L71

**Reply by jonmmease** [reply]
Thanks for the context, and no worries if you're not around to think about this more today!

**Reply by jonmmease** [reply]
not sure why we didn't need it before, but adding `raw_enabled = True` to `conf.py` fixed the above.

**Comment by dangotbanned** [comment]
@jonmmease I wanted to create another thread for this so it didn't get lost.

> I finished the release notes at [hackmd.io/@jEkhP0lORMCL8z1MH2P2SA/S188R-0zkg](https://hackmd.io/@jEkhP0lORMCL8z1MH2P2SA/S188R-0zkg) 

I'm really impressed, thank you for working on this 🚀

**Reply by mattijn** [reply]
Appreciate being mentioned too😉. I tried a few things but was not able to get into edit modus, but fine as is.

**Reply by jonmmease** [reply]
Oh, I just checked the share settings again and you weren't there any more. I re-added you (the account using your GitHub profile picture).

**Reply by mattijn** [reply]
Thanks! I added some examples to the release notes, I think it is auto-saved?

**Reply by jonmmease** [reply]
thanks for adding the examples! They look great

**Reply by mattijn** [reply]
Added an animated GIF as a thank you for the many PRs from @dangotbanned🥳

**Reply by dangotbanned** [reply]
Thank you @mattijn!

**Reply by dangotbanned** [reply]
![image](https://github.com/user-attachments/assets/2ee5c93a-5ef2-4bcd-839e-99f4019d80a0)

**Reply by dangotbanned** [reply]
I've added some comments on the release notes - really great work @mattijn, @jonmmease

**Reply by jonmmease** [reply]
Thanks! It seems, like with Mattijn, I needed to re-add you after you created an account.  So you should have write access now, but I'm happy to roll in your comments after I finish up the docs update.

**Reply by jonmmease** [reply]
I incorporated @dangotbanned's comments, so that the release notes are in final form from my side.

**Comment by dangotbanned** [comment]
We may want to update the pin for `narwhals` as @MarcoGorelli fixed https://github.com/narwhals-dev/narwhals/issues/1426

https://github.com/narwhals-dev/narwhals/releases/tag/v1.14.2

**Reply by jonmmease** [reply]
Were there any imports you wanted to change in altair with an update to narwhals 1.14.2? I don't see us importing from `narwhals.dependencies` anywhere right now.

**Reply by dangotbanned** [reply]
@jonmmease all done now in https://github.com/vega/altair/pull/3697

**Comment by jonmmease** [comment]
Ok, release PR is up at https://github.com/vega/altair/pull/3699. I plan to merge and publish the release in about an hour

**Reply by dangotbanned** [reply]
@jonmmease I've left some comments and [my diff](https://github.com/vega/altair/pull/3699#issuecomment-2495651459) on #3699, but might not be reachable later

Thank you for working on this 😄

**Reply by jonmmease** [reply]
thanks! taking a look now

**Comment by jonmmease** [comment]
Release is out! See https://github.com/vega/altair/releases/tag/v5.5.0. PyPI packages are published, the docs have been updated, and conda-forge packages are building now.

To other maintainers, feel free to review and edit the release notes.

@dangotbanned, feel free to make any additional changes or cleanup to doc build pipeline if you'd like things structured differently than how I handled it in https://github.com/vega/altair/pull/3699.

**Reply by mattijn** [reply]
Thanks @jonmmease for doing this release!

**Comment by dangotbanned** [comment]
@mattijn @jonmmease Just noticed the example for `Chart.transform_filter` might be incorrect:


![image](https://github.com/user-attachments/assets/f61d3da6-a5fc-461b-9014-001b1634a678)


It looks like the image is for:
> When passing multiple predicates they will be reduced with &

But the **Release Notes** code implies that:
- **Only** `year == 2000`
	- The legend is showing `1990`, `2000`
- **Only** `sex == 1` (Can't remember what this encoded, but the other was `2`)
- The full `age` range should be present
	- `90` is missing

**Reply by mattijn** [reply]
Fixed

**Reply by dangotbanned** [reply]
> Fixed

Thanks @mattijn
